modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/TaskActionExecutor.java

```diff
@@ -231,19 +239,47 @@ private boolean callActionServiceMethod(ActionEvent action, Map<String, Object>
             try {
                 Method method = service.getClass().getMethod(serviceMethod, methodHandler.getClasses());
 
-                try {
-                    method.invoke(service, methodHandler.getObjects());
-                } catch (IllegalAccessException | InvocationTargetException e) {
-                    throw new TaskHandlerException(
-                            ACTION, "task.error.serviceMethodInvokeError", e,
-                            serviceMethod, action.getServiceInterface()
-                    );
+                int numberOfRetries = task.getNumberOfRetries();
+                int retryIntervalTime = task.getRetryIntervalInMilliseconds();
+                boolean success;
+
+                for (int i = 0; i <= numberOfRetries; i++) {
+                    success = true;
+                    try {
+                        method.invoke(service, methodHandler.getObjects());
+                    } catch (IllegalAccessException | InvocationTargetException e) {
+                        success = false;
+                        if (i == numberOfRetries) {
+                            if (numberOfRetries == 0) {
+                                LOGGER.info("Failed action {} in {} not retry, because number of retries for task {} is 0",
+                                        serviceMethod, action.getServiceInterface(), task.getName());
+                            } else {
+                                LOGGER.info("Max retry count = {} reached for {} in {}", numberOfRetries, serviceMethod,
+                                        action.getServiceInterface());
+                            }
+                            throw new TaskHandlerException(
+                                    ACTION, "task.error.serviceMethodInvokeError", e,
+                                    serviceMethod, action.getServiceInterface()
+                            );
+                        } else if (retryIntervalTime != 0) {
+                            Thread.sleep(retryIntervalTime);
```

Comment:
This is bad, since we don't know the interval - it can be hours. A solution to this was discussed on the mailing list - we were supposed to use the scheduler through an event triggered job.